WeakAuras/GenericTrigger.lua

```diff
@@ -1582,6 +1582,38 @@ do
       mainSpeed, offSpeed = mainSpeedNew, offSpeedNew
     elseif casting and (event == "UNIT_SPELLCAST_INTERRUPTED" or event == "UNIT_SPELLCAST_FAILED") then
       casting = false
+    elseif event == "PLAYER_EQUIPMENT_CHANGED" and isAttacking then
+      local currentTime = GetTime()
+      mainSpeed, offSpeed = UnitAttackSpeed("player")
+      offSpeed = offSpeed or 0
+      local startEvent = false
+      if lastSwingMain then
+        lastSwingMain = currentTime
+        swingDurationMain = mainSpeed
+        mainSwingOffset = 0
+        timer:CancelTimer(mainTimer)
+        mainTimer = timer:ScheduleTimerFixed(swingEnd, mainSpeed, "main")
+        startEvent = true
+      end
+      if lastSwingOff then
+        lastSwingOff = currentTime
+        swingDurationOff = offSpeed
+        mainSwingOffset = 0
```

Comment:
Is setting mainSwingOffset to zero correct here?

Reply:
oops